Lib/fontTools/feaLib/ast.py

```diff
@@ -1043,24 +1266,34 @@ def asFea(self, indent=""):
 
 
 class ScriptStatement(Statement):
+    """A ``script`` statement."""
     def __init__(self, script, location=None):
         Statement.__init__(self, location)
-        self.script = script
+        self.script = script #: the script code
 
     def build(self, builder):
+        """Calls the builder's ``set_script`` callback."""
         builder.set_script(self.location, self.script)
 
     def asFea(self, indent=""):
         return "script {};".format(self.script.strip())
 
 
 class SinglePosStatement(Statement):
+    """A single position statement. ``prefix`` and ``suffix`` should be
+    lists of `glyph-containing objects`_.
+
+    ``pos`` should be a list of (`glyph-containing object`_,
+    :class:`ValueRecord`) tuples. There should probably be only one entry in
+    that list."""
```

Comment:
This should probably be more clear. What happens if there isn’t only one entry? Is it a violation of the API? Is it supported but just a silly thing to do?

Reply:
I'll clarify this, but separately we should add an assertion at this point, because currently a user could produce invalid output. Clearly it's not a Single Positioning rule if you have more than one entry in the list:

```
>>> SinglePosStatement([ ( GlyphName("a"), ValueRecord(1) ), ( GlyphName("b"), ValueRecord(2) ), ( GlyphName("c"), ValueRecord(3) )], [], [], False).asFea()
'pos a <1 0 0 0> b <2 0 0 0> c <3 0 0 0>;'
```